mirrord/kube/src/api/container.rs

```diff
@@ -104,23 +104,26 @@ where
         P: Progress + Send + Sync;
 }
 
-/// Choose container logic:
-///
-/// 1. Try to find based on given name
-/// 2. Try to find first container in pod that isn't a mesh sidecar
-/// 3. Take first container in pod
-///
-/// We also check if we're in a mesh based on `MESH_LIST`, returning whether we are or not.
 #[tracing::instrument(level = "trace", ret)]
-pub fn choose_container<'a>(
-    container_name: Option<&str>,
-    container_statuses: &'a [ContainerStatus],
-) -> (Option<&'a ContainerStatus>, Option<MeshVendor>, bool) {
+pub fn check_mesh_vendor(pod: &Pod) -> Option<MeshVendor> {
     const ISTIO: [&str; 2] = ["istio-proxy", "istio-init"];
     const LINKERD: [&str; 2] = ["linkerd-proxy", "linkerd-init"];
     const KUMA: [&str; 2] = ["kuma-sidecar", "kuma-init"];
 
-    let mesh = container_statuses.iter().find_map(|status| {
+    if pod
+        .metadata
+        .annotations
+        .as_ref()
+        .and_then(|annotations| annotations.get("ambient.istio.io/redirection"))
+        .map(|annotation| annotation == "enabled")
+        .unwrap_or_default()
+    {
+        return Some(MeshVendor::IstioAmbient);
+    }
```

Comment:
Now that we're actually creating `IstioAmbient` variant and passing it to agent CLI, there's a problem - what if the agent is older and doesn't recognize this variant?

Reply:
I though about it though I'm not sure how to solve it elegantly (also older version of agent won't work properly on mesh proxy anyway no?)

Reply:
Ok I did a compatibility change